src/Symfony/Component/HttpKernel/HttpCache/Store.php

```diff
@@ -106,7 +106,10 @@ public function unlock(Request $request)
 
     public function isLocked(Request $request)
     {
-        return is_file($this->getPath($this->getCacheKey($request).'.lck'));
+        $path = $this->getPath($this->getCacheKey($request).'.lck');
+        clearstatcache(true, $path);
+
+        return is_file($path);
```

Comment:
@MatTheCat is right here, only the information about non-existent files is not cached:

> You should also note that PHP doesn't cache information about non-existent files. So, if you call file_exists() on a file that doesn't exist, it will return FALSE until you create the file. If you create the file, it will return TRUE even if you then delete the file. However unlink() clears the cache automatically.